File: peek_plugin_tutorial/_private/tuples/StringIntDecreaseActionTuple.py
from vortex.Tuple import addTupleType, TupleField
from vortex.TupleAction import TupleActionABC

from peek_plugin_tutorial._private.PluginNames import tutorialTuplePrefix


@addTupleType
class StringIntDecreaseActionTuple(TupleActionABC):
    __tupleType__ = tutorialTuplePrefix + "StringIntDecreaseActionTuple"

    stringIntId = TupleField()
    offset = TupleField()

File: peek_plugin_tutorial/plugin-module/_private/tuples/StringIntDecreaseActionTuple.ts
                                         import {addTupleType, Tuple, TupleActionABC} from "@synerty/vortexjs";
import {tutorialTuplePrefix} from "../PluginNames";

@addTupleType
export class StringIntDecreaseActionTuple extends TupleActionABC {
    static readonly tupleName = tutorialTuplePrefix + "StringIntDecreaseActionTuple";

    stringIntId: number;
    offset: number;

    constructor() {
        super(StringIntDecreaseActionTuple.tupleName)
    }
}